src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -294,10 +355,100 @@ public override async ValueTask<IReadOnlyList<BulkStateItem>> GetBulkStateAsync(
                 Message = invokeRequest,
             };
 
-            return await this.MakeGrpcCallHandleError(
-                 options => client.InvokeServiceAsync(request, options),
-                 headers,
-                 cancellationToken);
+            var callOptions = new CallOptions(headers: headers ?? new Metadata(), cancellationToken: cancellationToken);
+
+            // add token for dapr api token based authentication
+            var daprApiToken = Environment.GetEnvironmentVariable("DAPR_API_TOKEN");
+
+            if (daprApiToken != null)
+            {
+                callOptions.Headers.Add("dapr-api-token", daprApiToken);
+            }
+
+            // Common Exception Handling logic can be added here for all calls.
+            var grpcCall = client.InvokeServiceAsync(request, callOptions);
+            return grpcCall;
```

Comment:
I would really strongly suggest refactoring this code path.

Calling an async method without immediately awaiting the result has some significant gotchas - because exceptions *might* be thrown synchronously or asynchronously - and that could change without  being considered a breaking change (and break you). 

It's better if you could have `MakeInvokeRequest` return the values of `request` and `callOptions` and call `client.InvokeServiceAsync` where your error handling code is.